### ISSUE TITLE
Question: "0.5.0 Native networked server mode with authentication."

### ISSUE THREAD

**Issue by den1k** [juji-io/datalevin#61]
1. Does this mean the DB would run on it's own server and could be queried / transacted to over http?
2. If so, does this lean towards a peer architecture similar to datomic on prem?

**Comment by huahaiy** [comment]
1. When running in server mode, Datalevin would run as a multi-tenant database server, just like Postgres, MySQL, mongodb, or most other conventional DB servers. The client (the Datalevin library itself, or some 3rd party code) will communicate with the server over a wire protocol in transit+json format, since we are already doing that when Datalevin runs as a babashka pod. It will not be over http as it incurs unnecessary overhead. See also https://github.com/juji-io/datalevin/blob/server/doc/dtlv.md#server

2. This will be a conventional client/server architecture. However, the client (the Datalevin library itself) can run its own DBs, so the local and remote DBs can be cross-queried in the same query as usual. In such cases, the query is running similar to peer architecture to some extent, as the query processing is mainly done locally. 
 
Basically, I would like Datalevin to be more like a conventional database, just be more versatile, flexible and convenient. There are many things I like about Datomic, but peer architecture is not one of them.

**Comment by den1k** [comment]
Great looking forward to this!

I assume this would break rich types like `d/entity` because transit does not support it? Or would the metadata needed for entity be somehow encoded during serialization in transit and re-instantiated during deserialization for Entity lookups?

**Comment by huahaiy** [comment]
Yes, I think we will try to encode/decode necessary info.

**Comment by huahaiy** [comment]
`entity` and friends will be running locally, only the storage layer and below is running on remote server, i.e. only the `.-store` attribute of a `DB` will be replaced by a proxy store that communicates with the remote server.

**Comment by huahaiy** [comment]
Done.
